### ISSUE TITLE
container networks aren't added in defined order

### ISSUE THREAD

**Issue by heidricha** [docker/compose#4083]
for a service with more then one network, the networks are created in a strange way:

eth0 / eth1 order is ok, BUT
eth0 is eth0@ifxxx and eth1 is eth1@ifyyy where yyy < xxx, so f.e. "ip addr" command displays the network interfaces like:
eth1
eth0
which is strange, but I guess doesn't harm at all.

Unfortunately /etc/hosts contains "hostname" twice with the addresses of the host in net1 and net2 accordingly AND the order of these lines in /etc/hosts is not fixed, for some services the order is
hostname eth1-IP
hostname eth0-IP

and vice versa for some services. I think it should be deterministic.

**Comment by shin-** [comment]
I'm not sure that's something we have control over at our level (the engine itself applies changes to `/etc/hosts`). Do you mind expanding on why this is an issue for you?

**Comment by heidricha** [comment]
Some silly application relies on how $(hostname) resolves. Since all network interfaces imply a line in /etc/hosts in a random order, the application get random IP addresses.

Before docker compose I used to start the whole stack with scripts. Running a container (by CLI) only allows me to add a single network, so all others should have been added later, so the order was fixed. Also I didn't use links. For some reason, the engine didn't mess up my etc/hosts.

Anyway, you are right, its a basic engine feature, I have to go there with my problem (meanwhile search for an application configuration based solution)
